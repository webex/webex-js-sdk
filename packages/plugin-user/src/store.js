/**!
 *
 * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
 */

import {patterns} from '@ciscospark/common';

const usersByEmail = new WeakMap();
const usersById = new WeakMap();

export default class UserStore {
  constructor() {
    usersByEmail.set(this, new Map());
    usersById.set(this, new Map());
  }

  add(user) {
    if (!user.id) {
      return Promise.reject(`\`user.id\` is required`);
    }

    if (!user.emailAddress) {
      return Promise.reject(`\`user.emailAddress\` is required`);
    }

    const p1 = this.getById(user.id)
      .then((u) => usersById.get(this).set(user.id, Object.assign({}, u, user)))
      .catch(() => usersById.get(this).set(user.id, Object.assign({}, user)));

    const p2 = this.getByEmail(user.emailAddress)
      .then((u) => usersByEmail.get(this).set(user.emailAddress, Object.assign({}, u, user)))
      .catch(() => usersByEmail.get(this).set(user.emailAddress, Object.assign({}, user)));

    return Promise.all([p1, p2]);
  }

  get(id) {
    if (patterns.uuid.test(id)) {
      return this.getById(id);
    }

    if (patterns.email.test(id)) {
      return this.getByEmail(id);
    }

    return Promise.reject(`\`id\` does not appear to be a valid user identifier`);
  }

  getById(id) {
    const ret = usersById.get(this).get(id);
    if (ret) {
      return Promise.resolve(ret);
    }
    return Promise.reject(new Error(`No user found identified by specified identifier`));
  }

  getByEmail(email) {
    const ret = usersByEmail.get(this).get(email);
    if (ret) {
      return Promise.resolve(ret);
    }
    return Promise.reject(new Error(`No user found identified by specified identifier`));
  }
}
