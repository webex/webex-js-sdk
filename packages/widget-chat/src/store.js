import {
  applyMiddleware,
  combineReducers,
  compose,
  createStore
} from 'redux';

import thunk from 'redux-thunk';
import createLogger from 'redux-logger';
import {
  activity,
  conversation,
  flags,
  indicators,
  notifications,
  share,
  user,
  widget
} from './reducers';
import sparkReducer from './modules/redux-spark/reducers';

const devtools = window.devToolsExtension || (() => (noop) => noop);

const enhancers = [
  applyMiddleware(
    thunk,
    createLogger({
      level: `info`,
      duration: true,
      collapsed: false,
      logErrors: process.env.NODE_ENV === `production`
    })
  ),
  devtools()
];

export default createStore(
  combineReducers({
    activity,
    conversation,
    flags,
    indicators,
    notifications,
    share,
    user,
    widget,
    spark: sparkReducer
  }),
  compose(...enhancers)
);
