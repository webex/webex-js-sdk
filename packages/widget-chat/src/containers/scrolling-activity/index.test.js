import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';

import store from '../../store';

import ScrollingActivity from '.';

const flags = [];
const activities = [{
  id: `test-123-123-123-123`,
  actor: {
    id: 1,
    displayName: `Test User 1`
  },
  object: {
    displayName: `Test Activity Content 1`
  },
  published: `2016-09-20T19:52:57.186Z`,
  verb: `post`
}, {
  id: `test-456-123-456-123`,
  actor: {
    id: 2,
    displayName: `Test User 2`
  },
  object: {
    displayName: `Test Activity Content 2`
  },
  published: `2016-09-20T19:53:57.186Z`,
  verb: `post`
}, {
  id: `test-789-123-789-123`,
  actor: {
    id: 3,
    displayName: `Test User 3`
  },
  object: {
    displayName: `Test Activity Content 3`
  },
  published: `2016-09-20T19:54:57.186Z`,
  verb: `post`
}];
const avatars = {
  'test-123-123-123-123': `http://googledoodle.jpg`
};
const onActivityDelete = jest.fn();
const onActivityFlag = jest.fn();

describe(`ScrollingActivity container`, () => {
  it(`renders properly`, () => {
    const component = renderer.create(
      <Provider store={store}>
        <ScrollingActivity
          activities={activities}
          avatars={avatars}
          flags={flags}
          onActivityDelete={onActivityDelete}
          onActivityFlag={onActivityFlag}
        />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
