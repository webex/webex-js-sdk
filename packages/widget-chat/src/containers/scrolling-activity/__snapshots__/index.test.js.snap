exports[`ScrollingActivity container renders properly 1`] = `
<div
  className={undefined}
  onScroll={undefined}>
  <div>
    <div
      className="activity-list">
      <div
        className="activity-item-container">
        <div
          className="activity-item">
          <div
            className="avatar-wrapper">
            <div
              className="avatar"
              style={undefined}>
              <span
                className="avatar-letter">
                T
              </span>
            </div>
          </div>
          <div
            className="content-container">
            <div
              className="meta">
              <div
                className="display-name"
                title="{name}">
                Test User 1
              </div>
              <div
                className="published">
                9/20/16, 3:52 PM
              </div>
            </div>
            <div
              className="activity-content">
              <div
                className="activity-text"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Test Activity Content 1",
                  }
                } />
            </div>
          </div>
          <div
            className="activity-post-actions">
            <div
              className="activity-post-action">
              <div
                className="post-action-item">
                <button
                  className="icon-button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Flag this message" />
                </button>
              </div>
            </div>
            <div
              className="activity-post-action">
              <div
                className="action-spacer" />
            </div>
          </div>
        </div>
      </div>
      <div
        className="activity-item-container">
        <div
          className="activity-item">
          <div
            className="avatar-wrapper">
            <div
              className="avatar"
              style={undefined}>
              <span
                className="avatar-letter">
                T
              </span>
            </div>
          </div>
          <div
            className="content-container">
            <div
              className="meta">
              <div
                className="display-name"
                title="{name}">
                Test User 2
              </div>
              <div
                className="published">
                9/20/16, 3:53 PM
              </div>
            </div>
            <div
              className="activity-content">
              <div
                className="activity-text"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Test Activity Content 2",
                  }
                } />
            </div>
          </div>
          <div
            className="activity-post-actions">
            <div
              className="activity-post-action">
              <div
                className="post-action-item">
                <button
                  className="icon-button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Flag this message" />
                </button>
              </div>
            </div>
            <div
              className="activity-post-action">
              <div
                className="action-spacer" />
            </div>
          </div>
        </div>
      </div>
      <div
        className="activity-item-container">
        <div
          className="activity-item">
          <div
            className="avatar-wrapper">
            <div
              className="avatar"
              style={undefined}>
              <span
                className="avatar-letter">
                T
              </span>
            </div>
          </div>
          <div
            className="content-container">
            <div
              className="meta">
              <div
                className="display-name"
                title="{name}">
                Test User 3
              </div>
              <div
                className="published">
                9/20/16, 3:54 PM
              </div>
            </div>
            <div
              className="activity-content">
              <div
                className="activity-text"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Test Activity Content 3",
                  }
                } />
            </div>
          </div>
          <div
            className="activity-post-actions">
            <div
              className="activity-post-action">
              <div
                className="post-action-item">
                <button
                  className="icon-button"
                  onClick={[Function]}
                  onKeyUp={[Function]}>
                  <div
                    className="icon"
                    title="Flag this message" />
                </button>
              </div>
            </div>
            <div
              className="activity-post-action">
              <div
                className="action-spacer" />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="indicators">
      <div />
    </div>
  </div>
</div>
`;
